core/amber/src/main/scala/edu/uci/ics/amber/engine/common/IOperatorExecutor.scala

```diff
@@ -13,7 +15,9 @@ trait IOperatorExecutor {
 
   def processTuple(
       tuple: Either[ITuple, InputExhausted],
-      input: LinkIdentity
+      input: LinkIdentity,
```

Comment:
If it's easy, can we refactor these 3 into a class called `ExecutorContext`? then expose setter and getter APIs.